docs/pages/database-access/guides/postgres-cloudsql.mdx

```diff
@@ -0,0 +1,336 @@
+
+Assign it the "Service Account Token Creator" role:
+
+![Grant Service Account Token Creator to Service Account](../../../img/database-access/guides/cloudsql/service-account-db-service-grant.png)
+
+Once created, go to that service account's Keys tab and create a new key:
+
+![Service Account Keys](../../../img/database-access/guides/cloudsql/service-account-keys.png)
+
+Make sure to choose JSON format:
+
+![Service Account New Key](../../../img/database-access/guides/cloudsql/service-account-new-key.png)
+
+Save the file, your Teleport Database Service will need it to be able to generate
+IAM auth tokens.
+
+## Gather Cloud SQL Instance Information
+
+To connect a Cloud SQL database to Teleport, you'll need to gather a few pieces
+of information about the instance.
+
+- GCP Project ID.
+
+You can normally see it in the organization view at the top of the GCP dashboard.
+
+- Cloud SQL instance ID.
+
+The instance ID is the name of your Cloud SQL instance shown at the top of the
+Overview page:
+
+![Instance ID](../../../img/database-access/guides/cloudsql/instance-id.png)
+
+- Cloud SQL instance endpoint.
+
+You will use the instance's public IP address to connect to it. It can be viewed
+on the "Connect to this instance" panel on the Overview page:
+
+![Instance Public IP](../../../img/database-access/guides/cloudsql/instance-public-ip.png)
+
+- Cloud SQL instance root certificate.
+
+The instance's root certificate is required so Teleport can validate the
+certificate presented by the database instance. You can download `server-ca.pem`
+file from the Connections tab under Security section:
+
+![Instance Root Certificate](../../../img/database-access/guides/cloudsql/instance-root-ca.png)
+
+## Configure Teleport
+
+Cloud SQL support for Teleport Database Access is available starting from
+`6.1` release.
+
+Download the appropriate version of Teleport for your platform from
+our [downloads page](https://goteleport.com/teleport/download).
+
+Follow the installation [instructions](../../installation.mdx).
+
+### Start Auth/Proxy Service
+
+Create a configuration file for a Teleport service that will be running
+auth and proxy servers:
+
+```yaml
+teleport:
+  data_dir: /var/lib/teleport
+  nodename: test
+auth_service:
+  enabled: "yes"
+proxy_service:
+  enabled: "yes"
+  # Set public address proxy will be reachable at.
+  public_addr: teleport.example.com:3080
+  # The PostgreSQL proxy listens on the regular web proxy port.
+  web_listen_addr: 0.0.0.0:3080
+ssh_service:
+  enabled: "no"
+```
+
+Start the service:
+
+```sh
+$ teleport start --config=/path/to/teleport.yaml
+```
+
+Generate a short-lived join token for the database service and save it for
+example in `/tmp/token`:
+
+```sh
+$ tctl tokens add \
+    --type=db \
+    --db-name=test \
+    --db-protocol=postgres \
+    --db-uri=35.1.2.3:5432
+```
+
+### Create Role and User
+
+Create a role that will allow a user to connect to any database using any
+database account:
```

Comment:
Can we use the `access` role?

Reply:
@benarent Do we have any documentation for these new built-in roles yet? I couldn't find it the other day by searching "editor" or "access" using the Algolia search.

Reply:
No, access role doesn't have wildcard db_names and db_users.

Reply:
Could we educate people for how to add them? ( If the below is correct? I just guessed back on `logins`) 

``` 
tctl --config=/path/to/teleport.yaml users add --roles=editor,access,db --db_name=foo --db_users=bar teleport-admin 



Access Role
```
kind: role
metadata:
  description: Access cluster resources
  id: 1617830297136031551
  name: access
spec:
  allow:
    app_labels:
      '*': '*'
    db_labels:
      '*': '*'
    db_names:
    - '{{internal.db_names}}'
    db_users:
    - '{{internal.db_users}}'
 ```

Reply:
@benarent Sorry not following, to add what? Access role is a part of default distribution but it just restricts db_names and db_users to `{{internal.db_names}}` as you pointed out above. If you mean to add these traits to the user, I'm not sure there's a command to do this?

Reply:
@r0mant I agree with Ben on access roles, they are documented here:

https://goteleport.com/docs/access-controls/getting-started/#step-13-add-local-users-with-preset-roles

Reply:
@klizhentas @benarent Yes, access role would be nice here but only allows access to `{{internal.db_names/users}}` which are empty for a local user so effectively "access" role doesn't give any access to any databases.

We need some way to update traits for a local use to set those db_names/db_users. Do you folks know if there's a way to update local user traits now? I couldn't find a good way to do it.

Should we bring back flags `--db-users` and `--db-names` for "tctl users add/update" commands which would set those respective traits?

Reply:
Nevermind, I found a pretty good way to do it. Updated.